client/client.go

```diff
@@ -528,12 +578,16 @@ func (r *NotaryRepository) Publish() error {
 		updatedFiles[data.CanonicalRootRole] = rootJSON
 	}
 
-	// we will always re-sign targets
-	targetsJSON, err := serializeCanonicalRole(r.tufRepo, data.CanonicalTargetsRole)
-	if err != nil {
-		return err
+	// iterate through all the targets files - if they are dirty, sign and update
+	for roleName, roleObj := range r.tufRepo.Targets {
```

Comment:
Note to reviewer:  I am slightly worried about no longer sending targets all the time - I *think* it should be fine, and all the tests seem to pass, but would appreciate it if folks took a harder look at this.